src/Lti/LoadBalancer/UserGroupIdLtiInstanceLoadBalancer.php

```diff
@@ -58,4 +59,32 @@ public function getLtiRequestContextId(Assignment $assignment): string
 
         return (string)$user->getGroupId();
     }
+
+    public function generateGroupIds(string $groupPrefix): array
+    {
+        $totalInstances = $this->ltiInstanceCollection->count();
+        if ($this->ltiInstanceCollection->isEmpty()) {
+            throw new LtiInstanceNotFoundException('No Lti instance were found in database.');
+        }
+
+        $targetId = 1;
+        $groupIds = [];
+        while ($targetId <= $totalInstances) {
+            $newGroupId = sprintf($groupPrefix . '_%s', substr(md5(random_bytes(10)), 0, 10));
+            $possibleIndex = (int) $this->computeLtiInstanceByString($newGroupId)->getId();
```

Comment:
what is a retutn type of ->getId() ?

Reply:
It should be null or int

Reply:
in this case it is cheaper to add `null !== $possibleIndex &&` to the if condition then cast null to int and then check if 0 index exists